### ISSUE TITLE
No module named 'network' when enabling

### ISSUE THREAD

**Issue by dynjo** [daniloc/GeometryLink#1]
<img width="647" alt="image" src="https://github.com/daniloc/GeometryLink/assets/184405/8e7910a4-60d3-4dac-8f7b-4fc806f7dee3">

**Comment by dynjo** [comment]
I solved this by doing:

`/Applications/Blender.app/Contents/Resources/4.0/python/bin/python3.10 -m pip install network`

**Comment by daniloc** [comment]
I'm not sure that's going to solve it for you, @dynjo. Is it possible you tried to install the folder instead of the zip file into Blender?

**Comment by dynjo** [comment]
I did install the zip, but as soon as I clicked the enable checkbox in Blender prefs it threw the error. Manually installing network did solve that, but sadly still no connection to my AVP, so maybe another issue.

**Comment by daniloc** [comment]
@dynjo Remove the add-on in Blender, then in Finder go to folder:

/Users/jan/Library/Application Support/Blender/4.0/scripts/addons/

Make sure "Blender Plugin" is not there. Then reinstall the add-on using the zip file again.

At that point, anything interesting in the logs when you switch over to the Scripting tab?

I just tried installing the zip from scratch and it gave me no trouble, so I'm hoping the logs can point us to where the hitch is.

**Comment by daniloc** [comment]
I don't know much about python, especially as it relates to naming collisions, but it also occurs to me that having pip remove the other "network" module you installed might help too.

**Comment by dynjo** [comment]
Yeah sorry, you are of course correct, me installing `network` was just hiding the issue, which appears to be that my Blender Python install cannot resolve your `network.py` in the `Geometry Link` directory.

Also not a python expert, but I will keep fiddling with it 👍🏼

**Comment by daniloc** [comment]
@dynjo Grateful for the report! Keep me posted, this is the most brittle part of the whole affair so I'm eager to learn anything I can about it

**Comment by justbcuz** [comment]
@dynjo Were you able to resolve this? I am experiencing the exact same issue.

![Screenshot 2024-04-12 at 1 00 45 PM](https://github.com/daniloc/GeometryLink/assets/186479/0a62127e-aec9-45c7-9ca6-84708e4b83c5)

**Comment by dynjo** [comment]
Sadly I didn't, sorry.

**Comment by BrianMila** [comment]
[Blender Plugin.zip](https://github.com/daniloc/GeometryLink/files/15190005/Blender.Plugin.zip)

@dynjo @justbcuz I also had the same problem.  I was sure it was a pathing issue, I messed with it for several hours with no success, and just before I gave up, I found the solution!  It WAS a pathing issue, and my guess is that it actually wasn't working for @daniloc but maybe Blender found the source files in some path, but not from the zip. I have no idea how to send a PR for a public repo, sorry...But for anyone who might be waiting, or in case it doesn't get updated, here is the top of __init.py__ that you need to make changes to, then zip up the Blender Plugin directory and install the zip as the plugin.  I've attached the updated Blender Plugin that works for me.

import os
import sys
import bpy
from bpy.app.handlers import depsgraph_update_post


local_module_path=os.path.join(os.path.dirname(os.path.abspath(__file__)),'Geometry Link')
if local_module_path not in sys.path:
    sys.path.append(local_module_path)

bl_info = {
    "name": "Geometry Link",
    "blender": (2, 80, 0),
    "category": "Object",
    "version": (1, 0, 0),
    "author": "Danilo Campos",
    "description": "Monitors geometry changes and serializes them to JSON",
}


import network
import os
import base64
import threading
